.github/workflows/release.yml

```diff
@@ -0,0 +1,225 @@
+          cache: 'maven'
+      - name: Prepare changelog for release
+        if: ${{ inputs.update_changelog }}
+        run: |
+          java .ci/ReleaseChangelog.java CHANGELOG.asciidoc ${{ inputs.version }}
+          git commit -m "Prepare changelog for release ${{ inputs.version }}" CHANGELOG.asciidoc
+      - name: Bump version and add git tag
+        run: ./mvnw release:prepare -B -DpushChanges=false "-Darguments=-DskipTests -Dmaven.javadoc.skip=true" -DreleaseVersion=${{ inputs.version }}
+      - run: git push --atomic origin ${{ inputs.branch }} v${{ inputs.version }}
+
+
+  maven_central_deploy:
+    name: "Deploy to Maven Central (Buildkite)"
+    runs-on: ubuntu-latest
+    needs:
+      - prepare_release
+
+    # TODO: Run the deployment on Buildkite
+    # Buildkite needs to
+    # - checkout the tag "v${{ inputs.version }}" (which is created by the prepare_release job)
+    # - run ./mvnw -s .ci/settings.xml -Pgpg clean deploy -DskipTests --batch-mode
+    # Note that the previous ./mvnw release:prepare release:perform from jenkins are not required anymore:
+    # prepare_release does all the fiddling with the git history
+    steps:
+      - run: echo "Not implemented yet!"
+
+  await_artifact_on_maven_central:
+    name: "Wait for artifacts to be available on maven central"
+    runs-on: ubuntu-latest
+    needs:
+      - maven_central_deploy
+    steps:
+      - uses: actions/checkout@v3
+      - shell: bash
+        run: |
+          until .ci/release/wait_maven_artifact_published.sh ${{ inputs.version }}
+          do
+            echo "Artifacts not found on maven central. Sleeping 30 seconds, retrying afterwards"
+            sleep 30s
+          done
+  
+
+  update_major_branch:
+    name: "Update Major Branch"
+    runs-on: ubuntu-latest
+    needs:
+      - await_artifact_on_maven_central
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+      - run: .ci/release/update_major_branch.sh ${{ inputs.version }}
+      - run: git push -f origin "$(echo '${{ inputs.version }}' | sed -E 's/\..+/.x/')"
+
+  update_cloudfoundry:
+    name: "Update Cloudfoundry"
+    runs-on: ubuntu-latest
+    needs:
+      - await_artifact_on_maven_central
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+      - name: "Update Cloudfoundry index.yml file"
+        shell: bash
+        run: .ci/release/update_cloudfoundry.sh ${{ inputs.version }}
+      - run: git push origin main
+
+
+  build_docker_images:
+    name: "Build and push docker images"
+    runs-on: ubuntu-latest
+    needs:
+      - await_artifact_on_maven_central
+    env:
+      TAG_NAME: v${{ inputs.version }}
+      SONATYPE_FALLBACK: 1
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: main
```

Comment:
Will this branch need to be the one defined in the inputs?

Reply:
https://github.com/elastic/apm-agent-java/pull/3137/commits/e976a8c019f1ed81c250394fc9aaeafa2db2a8a1